enhancements/network/allow-mtu-and-overlay-port-changes.md

```diff
@@ -0,0 +1,165 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Customers may need to change the MTU, or the ports used for vxlan or geneve
+tunnels post-installation. However these changes aren't trivial and may cause
+downtime, hence the CNO forbids currently them.
+
+We propose a brand new daemonset that will run on every node of the cluster and
+will make the necessary changes in an ordered, and coordinated manner with a
+service disruption within the TCP timeout.
+
+## Motivation
+
+While customers usually set the MTU, and the overlay ports some times they
+need to change them afterwards for unpredictable reasons such as performance,
+or changes in the underlay.
+
+## Goals
+
+* Allow to change MTU post install on both OpenShift SDN and OVN Kubernetes.
+
+* Allow to change the overlay network ports on the underlay in both OpenShift
+  SDN and OVN Kubernetes.
+
+* Allow to change both vxlan and geneve ports in a single operation.
+
+## Non goals
+
+* Allow to change MTU configuration in te underlay interfaces.
+
+* Allow to change MTU, and either port in a single operation.
+
+* Other safe or unsafe configuration changes.
+
+## Proposal
+
+When the cluster network operator detects the change it will deploy a new
+deaemonset with restartPolicy Mever which is responsible for doing the changes.
+
+Even though both changing the MTU, and changing the port have a lot in common,
+there are some key differences and therefore the daemonset will be slightly
+different.
+
+When decreasing the MTU each of the deamonset will:
+
+1. Verify that the new MTU is valid for that node. If it is valid add to
+   itself the annotation `openshift.io/cno-able-to-change-mtu`.
+2. Wait until the timestamp defined in `CHANGE_MTU_INIT_TS`
+3. Verify that every pod of that daemonset is running and has the annotation
+   `openshift.io/cno-able-to-change-mtu` if any of the pods is not running, or
+   is missing that annotation the pod will abort the procedure.
+4. Cordon the node where its running.
+5. Wait a grace period. This grace period is based on
+   `CHANGE_MTU_INIT_TS` rather than the time of finalization of the
+   previous step.
+6. Proceed to change the MTU of `eth0` on each one of the pods netnamespace.
+7. Wait a grace period. This grace period is based on
+   `CHANGE_MTU_INIT_TS` rather than the time of finalization of the
+   previous step.
+8. Proceed to change the MTU of each pod in the virtual switch.
+9. Wait a grace period. This grace period is based on
+   `CHANGE_MTU_INIT_TS` rather than the time of finalization of the
+   previous step.
+10. In the case of OVN Kubernetes proceed to change the MTU of the interfaces
+   ovn-k8s-mp0, br-local, and ovn-k8s-gw0 sequentially without any delay in
+   between.
+
+   In the case of OpenShiftSDN proceed to change the MTU of the interfaces
+   vxlansys.
+
```

Comment:
Do you mean e.g. `vxlan_sys_4789` ?  If so, we probably want to leave the vxlan and geneve interface MTU at 65000

```
4: genev_sys_6081: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 65000 qdisc noqueue master ovs-system state UNKNOWN group default qlen 1000
    link/ether 7e:a4:52:23:c4:e8 brd ff:ff:ff:ff:ff:ff
```
```
5: vxlan_sys_4789: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 65000 qdisc noqueue master ovs-system state UNKNOWN group default qlen 1000
    link/ether 2a:6f:6d:e9:0d:20 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::286f:6dff:fee9:d20/64 scope link
       valid_lft forever preferred_lft forever
```

Reply:
`vxlan_sys_4789` is a virtual interface that we shouldn't ever touch directly. You can look at the existing code to see what interfaces we set the MTU on at startup. (I think it's `br0` and `tun0`)

Reply:
I meant `vxlan_sys_4789`, anyway Dan's right and I removed this step as it's not necessary, I just realized it has a MTU of 65000 by default.